rcl/src/rcl/arguments.c

```diff
@@ -941,23 +936,12 @@ rcl_arguments_copy(
 
   rcl_allocator_t allocator = args->impl->allocator;
 
-  args_out->impl = allocator.allocate(sizeof(rcl_arguments_impl_t), allocator.state);
-  if (NULL == args_out->impl) {
-    return RCL_RET_BAD_ALLOC;
-  }
-
-  args_out->impl->allocator = allocator;
-
-  // Zero so it's safe to call rcl_arguments_fini() if an error occurrs while copying.
-  args_out->impl->num_remap_rules = 0;
-  args_out->impl->remap_rules = NULL;
-  args_out->impl->unparsed_args = NULL;
-  args_out->impl->num_unparsed_args = 0;
-  args_out->impl->unparsed_ros_args = NULL;
-  args_out->impl->num_unparsed_ros_args = 0;
-  args_out->impl->parameter_overrides = NULL;
-  args_out->impl->parameter_files = NULL;
-  args_out->impl->num_param_files_args = 0;
+  do {
```

Comment:
Any particular reason for this `do{ ... } while (false)` loop?

Reply:
Thank you for comment.
I wanted to narrow the scope of "ret".

If "ret" is defined in the function scope, it seems to be rcl_arguemnts_copy return value.
But there are lines which directly return values, ex `return RCL_RET_BAD_ALLOC;` or `return return RCL_RET_OK;`. I feel "why not return ret!?".

As I want to use "ret" very temporarily, so I use `do{ ..} while(false)`

How do you think these ideas?

(1) simply use "ret" (line 991 is also affected)
(2) change variable name such as "_ret".
(3) bellow

```
if (RCL_RET_OK != _rcl_allocate_initialized_arguments_impl(args_out, &allocator)) {
  return ret;
}
```

Reply:
The problem with 3) is that you won't have the correct value to return.  You could do assignment in the `if` statement, but I generally find that hard to read.

In this case, I think we just have to accept that this is a limitation of C and just leave `ret` as function scoped.  Since we are reducing the complexity of this function anyway, I don't think it will be hard to read.